cypress/support/commands.ts

```diff
@@ -0,0 +1,192 @@
+  cy.wrap(wiremock().requests.resetAllRequests());
+});
+
+Cypress.Commands.add("logRequests", () => {
+  cy.wrap(
+    wiremock()
+      .requests.getAllRequests()
+      .then((data) => {
+        data.requests.forEach((requestResponse) => {
+          const request = requestResponse.request;
+          Cypress.log({
+            name: "Wiremock",
+            message: `${request.method}: ${request.url}`,
+          });
+        });
+      })
+  );
+});
+
+Cypress.Commands.add("drupalLogin", () => {
+  cy.visit("/user/login");
+  cy.get('[name="name"]')
+    .type(Cypress.env("DRUPAL_USERNAME"))
+    .parent()
+    .get('[name="pass"]')
+    .type(Cypress.env("DRUPAL_USERNAME"));
+  cy.get('[value="Log ind"]').click();
+});
+
+Cypress.Commands.add("drupalLogout", () => {
+  cy.visit("/logout");
+});
+
+Cypress.Commands.add("drupalCron", () => {
+  // Because we run Wiremock as a proxy only services configured with the
+  //  proxy will use it. We need to proxy requests during cron and only the
+  // web container is configured to use the proxy and thus we have to run
+  // cron through the web frontend. Using the proxy with the CLI container would
+  // cause too many irrelevant requests to pass throuh the proxy.
+  cy.drupalLogin();
+  cy.visit("/admin/config/system/cron");
+  cy.get('[value="Kør cron"]').click();
+  cy.contains("Cron-opgave fuldført.");
+  cy.drupalLogout();
+});
+
+Cypress.Commands.add(
+  "adgangsplatformenLogin",
+  (authorizationCode: string, accessToken: string, userGuid: string) => {
+    cy.createMapping({
+      request: {
+        method: "GET",
+        urlPath: "/oauth/authorize",
+        queryParameters: {
+          response_type: {
+            equalTo: "code",
+          },
+        },
+      },
+      response: {
+        status: 302,
+        headers: {
+          location: `{{request.query.[redirect_uri]}}?code=${authorizationCode}&state={{request.query.[state]}}`,
+        },
+        transformers: ["response-template"],
+      },
+    });
+
+    cy.createMapping({
+      request: {
+        method: "POST",
+        urlPath: "/oauth/token/",
+        bodyPatterns: [
+          {
+            contains: "grant_type=authorization_code",
+          },
+          {
+            contains: `code=${authorizationCode}`,
+          },
+        ],
+      },
+      response: {
+        jsonBody: {
+          access_token: accessToken,
+          token_type: "Bearer",
+          expires_in: 2591999,
+        },
+      },
+    });
+
+    cy.createMapping({
+      request: {
+        method: "GET",
+        urlPath: "/userinfo/",
+        headers: {
+          Authorization: {
+            equalTo: `Bearer ${accessToken}`,
+          },
+        },
+      },
+      response: {
+        jsonBody: {
+          attributes: {
+            uniqueId: userGuid,
+          },
+        },
+      },
+    });
+
+    cy.visit("/user/login");
+    cy.contains("Log in with Adgangsplatformen").click();
```

Comment:
Is it deliberately that you do not return the `cy`object?
When the command is used in a test will the subsequent tests/assertions wait for this to finish?

Reply:
Ok. Great. Just wanted to challenge it :)